seisidd/experiment.py

```diff
@@ -401,16 +544,845 @@ def scan_closure():
             yield from self.collect_dark_field(cfg['tomo'])
     
         return (yield from scan_closure())
+    
+    #   summarize_plan with config yml file
+    def dryrun(self, scan_config):
+        """use summarize_plan for quick analysis"""
+        return summarize_plan(tomo_scan(self, scan_config))
 
 
 class NearField(Experiment):
-    """nf-HEDM control for 6-ID-D"""
-    pass
+    """NF-HEDM control for 6-ID-D"""
+    
+    def __init__(self, mode='debug'):
+        Experiment.__init__(self, mode)
+        self._mode = mode
+        # instantiate device
+        self.nf_stage       = NearField.get_nfstage(self._mode)
+        self.fly_control    = NearField.get_flycontrol(self._mode)
+        self.nf_det         = NearField.get_detector(self._mode)
+        self.nf_beam        = NearField.get_nfbeam(self._mode)
+        # TODO:
```

Comment:
Once everything is properly calibrated, we can rely on automated initialization during runtime.  However, I believe manual adjustment is unavoidable, and probably the best way to go for a long time.

Reply:
Let's wait for actual testing on this...